### PR TITLE
loopout: use psbt when obtaining the MuSig2 signature

### PR DESCRIPTION
**Pull request by bhandras**
This PR extends the `swapsserverrpc.MuSig2SignSweepReq` with the sweep txn's psbt to make server side validation of the sighash possible.

### CODE REVIEW

**Review by sputn1ck — APPROVED** [review]
Useful change! LGTM 🚀

File: sweep/sweeper.go [hunk 1]
```diff
@@ -20,51 +22,53 @@ type Sweeper struct {
...
-	sweepTx.AddTxOut(&wire.TxOut{
-		PkScript: sweepPkScript,
-		Value:    int64(amount - fee),
-	})
+	// Compose tx.
+	sweepTx := &wire.MsgTx{
+		Version: 2,
+		TxIn: []*wire.TxIn{{
+			PreviousOutPoint: htlcOutpoint,
+			SignatureScript:  htlc.SigScript,
```
Comment (guggero) [inline sweep/sweeper.go:48]: Are np2wkh HTLCs still supported? Otherwise we could scrap the `SignatureScript`.

File: sweep/sweeper.go [hunk 2]
```diff
@@ -20,51 +22,53 @@ type Sweeper struct {
...
+	}
 
-	// We need our previous outputs for taproot spends, and there's no
-	// harm including them for segwit v0, so we always include our prevOut.
-	prevOut := []*wire.TxOut{
-		{
-			Value:    int64(amount),
-			PkScript: htlc.PkScript,
-		},
+	packet, err := psbt.NewFromUnsignedTx(sweepTx)
```
Comment (guggero) [inline sweep/sweeper.go:57]: Should we set the `WitnessUtxo` field on the input as well?

**Review by guggero — APPROVED** [review]
LGTM :tada: 

I think just for completeness' sake we should add the UTXO information to the packet, even though it isn't strictly needed.

File: sweep/sweeper.go (hunk 2 above)
Comment (bhandras) [reply sweep/sweeper.go:56]: done

File: sweep/sweeper.go (hunk 1 above)
Comment (bhandras) [reply sweep/sweeper.go]: removed

**Pull request merged**